app/com/arpnetworking/metrics/portal/scheduling/JobMessageExtractor.java

```diff
@@ -25,18 +26,24 @@
  * @author Spencer Pearson (spencerpearson at dropbox dot com)
  */
 public final class JobMessageExtractor extends ShardRegion.HashCodeMessageExtractor {
+    private final JobRefSerializer _refSerializer;
+
     /**
      * Public constructor.
+     *
+     * @param refSerializer the serializer to use to construct the entity ID from each job ref.
      */
-    public JobMessageExtractor() {
+    @Inject
+    public JobMessageExtractor(final JobRefSerializer refSerializer) {
         super(NUM_SHARDS);
+        _refSerializer = refSerializer;
     }
 
     @Override
     @Nullable
     public String entityId(final Object message) {
         if (message instanceof JobExecutorActor.Reload) {
-            return jobRefToUId(((JobExecutorActor.Reload) message).getJobRef());
+            return _refSerializer.jobRefToEntityID(((JobExecutorActor.Reload) message).getJobRef());
```

Comment:
Is there any case where a reload would change the ref? I'm starting to think that a reload should be payload-less and just trigger the logic instead of describing what to reload. Effectively the latter changes the actor's identify without changing its name. Seems dangerous.

Reply:
Commented on most of this above.

I will agree that this is brittle, it relies on the actor being set up correctly by some external code yet in a way the actor implementation understands. The identity/name correspondence should *always* be correct if used via a ClusterSharding manager but there's nothing in the type system / code to enforce that invariant.